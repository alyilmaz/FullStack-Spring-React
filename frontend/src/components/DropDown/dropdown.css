 /* Status Dropdown */
 .custom-dropdown {
    position: relative;
    display: inline-block;
    vertical-align: middle;
    margin: 1px; 
    margin-left: 50px;
  }
  
  .custom-dropdown select {
    background-color: #fff;
    color: #000;
    font-size: 16px;
    padding: 8px;
    height: 40px;
    padding-right: 30px; 
    border: 1px solid rgb(0, 26, 255);
    margin: 0;
    border-radius: 5px;
    text-indent: 0.01px;
    text-overflow: '';
    -webkit-appearance: button; /* hide default arrow in chrome OSX */
  }
  
  .custom-dropdown::before,
  .custom-dropdown::after {
    content: "";
    position: absolute;
    pointer-events: none;
  }
  
  .custom-dropdown::after { 
    content: "\25BC";
    height: 1px;
    font-size: 10px;
    line-height: 1;
    right: 8px;
    top: 20%;
    margin-top: -2px;
  }
  
  .custom-dropdown::before { 
    width: 30px;
    right: 0;
    top: 0;
    bottom: 0;
    border-radius: 0 3px 3px 0;
  }
  
  .custom-dropdown select[disabled] {
    color: rgba(70, 155, 170, 0.3);
  }
  
  .custom-dropdown select[disabled]::after {
    color: rgba(24, 83, 122, 0.1);
  }
  
  .custom-dropdown::before {
    background-color: rgba(34, 133, 158, 0.15);
  }
  
  .custom-dropdown::after {
    color: rgba(24, 145, 129, 0.4);
  }