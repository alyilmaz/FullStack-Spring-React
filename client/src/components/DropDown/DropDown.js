import "./dropdown.css";

const DropDown = ({onChange, placeholder, options}) =>{

    return(
        <div>
            <span className="custom-dropdown">  
                <select id = "select" onChange={event => onChange(event.target.value)}>
                    <option key = "blank" value="" disabled >{placeholder !== undefined ? placeholder : "options"}</option>
                    {options?.map(op =>{
                        return(
                        <option key  = {op.id} value={op.id}>{op.label}</option>
                        )
                    })}           
                </select>
            </span> 
        </div>
    )
}

export default DropDown;