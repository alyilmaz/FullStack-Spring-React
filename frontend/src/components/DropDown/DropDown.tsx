import "./dropdown.css";

interface Props{
    placeholder: string,
    options: Options[],
    onChange(e:React.ChangeEvent<HTMLSelectElement>): void;
}

type Options = {
    id: string,
    label: string,
}

const DropDown = (props:Props) =>{

    return(
        <div>
            <span className="custom-dropdown">  
                <select id = "select" onChange={event => props.onChange(event)}>
                    <option key = "blank" value="" disabled >{props.placeholder !== undefined ? props.placeholder : "options"}</option>
                    {props.options?.map(op=>{
                        return(
                        <option key  = {op.id} value={op.id}>{op.label}</option>
                        )
                    })}           
                </select>
            </span> 
        </div>
    )
}

export default DropDown;