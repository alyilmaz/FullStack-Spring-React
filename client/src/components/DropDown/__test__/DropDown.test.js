/* eslint-disable testing-library/no-node-access */
import {render, screen} from "@testing-library/react";
import DropDown from "../DropDown";

const options = [{id:"all", label:"All"}, {id:"COMPLETED", label:"COMPLETED"},{id:"ERROR", label:"ERROR"},{id:"CANCELED", label:"CANCELED"}];

test("status option place holder should be options", () =>{
 
    render(<DropDown options={options}/>);
    expect(screen.getByText("options")).toBeTruthy();
})

test("status option place holder should be same the name of placeholder of props", () =>{
    const placeholder = "test";
    render(<DropDown placeholder={placeholder} options={options}/>);
    expect(screen.getByText(placeholder)).toBeTruthy();
})


test("there is only one option in select element if no options pros", () =>{
    const placeholder = "test";
    render(<DropDown placeholder={placeholder} />);
    expect(document.getElementById('select').childElementCount).toEqual(1);
})

test("the number of total options in select element should be size of options array(props) + 1", () =>{
    const placeholder = "test";
    const totatlChildElement = options.length + 1;
    render(<DropDown placeholder={placeholder} options={options}/>);
    expect(document.getElementById('select').childElementCount).toEqual(totatlChildElement);
})

test("all opitons should be enabled except status", () =>{
 
    render(<DropDown placeholder={"Status"} options={options}/>);
    expect(screen.getByText("Status")).toBeDisabled();
    expect(screen.getByText("All")).toBeEnabled();
    expect(screen.getByText("COMPLETED")).toBeEnabled();
    expect(screen.getByText("ERROR")).toBeEnabled();
    expect(screen.getByText("CANCELED")).toBeEnabled();
})