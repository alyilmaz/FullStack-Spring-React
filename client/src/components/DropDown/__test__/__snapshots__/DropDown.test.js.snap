// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snaphot 1`] = `
<div>
  <span
    className="custom-dropdown"
  >
    <select
      id="select"
      onChange={[Function]}
    >
      <option
        disabled={true}
        value=""
      >
        Status
      </option>
      <option
        value="all"
      >
        All
      </option>
      <option
        value="COMPLETED"
      >
        COMPLETED
      </option>
      <option
        value="ERROR"
      >
        ERROR
      </option>
      <option
        value="CANCELED"
      >
        CANCELED
      </option>
    </select>
  </span>
</div>
`;
